tests/test_filesystem.py

```diff
@@ -61,24 +61,61 @@ def test_get_file_interrupt(tmpdir, bad_file_system, backend):
     indirect=True,
 )
 @pytest.mark.parametrize(
-    'file, version, extensions, expected',
+    'file, version, extensions, regex, expected',
     [
-        ('/file.tar.gz', '1.0.0', None, 'file.tar/1.0.0/file.tar-1.0.0.gz'),
-        ('/file.tar.gz', '1.0.0', [], 'file.tar/1.0.0/file.tar-1.0.0.gz'),
-        ('/file.tar.gz', '1.0.0', ['tar.gz'], 'file/1.0.0/file-1.0.0.tar.gz'),
-        ('/.tar.gz', '1.0.0', ['tar.gz'], '.tar/1.0.0/.tar-1.0.0.gz'),
-        ('/tar.gz', '1.0.0', ['tar.gz'], 'tar/1.0.0/tar-1.0.0.gz'),
-        ('/.tar.gz', '1.0.0', None, '.tar/1.0.0/.tar-1.0.0.gz'),
-        ('/.tar', '1.0.0', None, '.tar/1.0.0/.tar-1.0.0'),
-        ('/tar', '1.0.0', None, 'tar/1.0.0/tar-1.0.0'),
+        (
+            '/file.tar.gz', '1.0.0', None, False,
+            'file.tar/1.0.0/file.tar-1.0.0.gz',
+        ),
+        (
+            '/file.tar.gz', '1.0.0', [], False,
+            'file.tar/1.0.0/file.tar-1.0.0.gz',
+        ),
+        (
+            '/file.tar.gz', '1.0.0', ['tar.gz'], False,
+            'file/1.0.0/file-1.0.0.tar.gz',
+        ),
+        (
+            '/.tar.gz', '1.0.0', ['tar.gz'], False,
+            '.tar/1.0.0/.tar-1.0.0.gz',
+        ),
+        (
+            '/tar.gz', '1.0.0', ['tar.gz'], False,
+            'tar/1.0.0/tar-1.0.0.gz',
+        ),
+        (
+            '/.tar.gz', '1.0.0', None, False,
+            '.tar/1.0.0/.tar-1.0.0.gz',
+        ),
+        (
+            '/.tar', '1.0.0', None, False,
+            '.tar/1.0.0/.tar-1.0.0',
+        ),
+        (
+            '/tar', '1.0.0', None, False,
+            'tar/1.0.0/tar-1.0.0',
+        ),
+        # test regex
+        (
+            '/file.0.tar.gz', '1.0.0', [r'\d+.tar.gz'], False,
+            'file.0.tar/1.0.0/file.0.tar-1.0.0.gz',
+        ),
+        (
+            '/file.0.tar.gz', '1.0.0', [r'\d+.tar.gz'], True,
+            'file/1.0.0/file-1.0.0.0.tar.gz',
+        ),
+        (
+            '/file.99.tar.gz', '1.0.0', [r'\d+.tar.gz'], True,
+            'file/1.0.0/file-1.0.0.99.tar.gz',
+        ),
```

Comment:
I would add a test for our actual use case:

```suggestion
        ),
        (
            '/file.prediction.99.tar.gz', '1.0.0', [r'prediction.\d+.tar.gz', r'truth.tar.gz'], True,
            'file/1.0.0/file-1.0.0.prediction.99.tar.gz',
        ),
        (
            '/file.truth.tar.gz', '1.0.0', [r'prediction.\d+.tar.gz', r'truth.tar.gz'], True,
            'file/1.0.0/file-1.0.0.truth.tar.gz',
        ),
```

What still does not work for me is to mix TAR.GZ with 99.TAR.GZ, but maybe I just define the wrong search pattern: `r'(\d.)?tar.gz'`?